edb/edgeql/compiler/setgen.py

```diff
@@ -780,6 +790,98 @@ def _is_computable_ptr(
     )
 
 
+def compile_enum_path(
+        expr: qlast.Path,
+        *,
+        source: s_types.Type,
+        ctx: context.ContextLevel) -> irast.Set:
+
+    assert isinstance(source, s_scalars.ScalarType)
+    enum_values = source.get_enum_values(ctx.env.schema)
+    assert enum_values is not None
+
+    nsteps = len(expr.steps)
+    if nsteps == 1:
+        raise errors.QueryError(
+            f"'{source.get_displayname(ctx.env.schema)}' enum "
+            f"path expression lacks an enum member name, as in "
+            f"'{source.get_displayname(ctx.env.schema)}.{enum_values[0]}'",
+            context=expr.steps[0].context,
+        )
+
+    step2 = expr.steps[1]
+    if not isinstance(step2, qlast.Ptr):
+        raise errors.QueryError(
+            f"an enum member name must follow enum type name in the path, "
+            f"as in "
+            f"'{source.get_displayname(ctx.env.schema)}.{enum_values[0]}'",
+            context=step2.context,
+        )
+
+    step2_direction = s_pointers.PointerDirection.Outbound
+    if step2.direction is not None:
+        step2_direction = s_pointers.PointerDirection(step2.direction)
+    if step2_direction is not s_pointers.PointerDirection.Outbound:
+        raise errors.QueryError(
+            f"enum types do not support backward link navigation",
+            context=step2.context,
+        )
+    if step2.type == 'property':
+        raise errors.QueryError(
+            f"enum types do not support link properties",
+            context=step2.context,
+        )
+
+    if nsteps > 2:
+        raise errors.QueryError(
+            f"enum types only support two-step paths: "
+            f"`enum name`.`member name`",
+            context=expr.steps[2].context,
+        )
+
+    return enum_indirection_set(
+        source=source,
+        ptr_name=step2.ptr.name,
+        source_context=expr.context,
+        ctx=ctx,
+    )
+
+
+def enum_indirection_set(
+        *,
+        source: s_types.Type,
+        ptr_name: str,
+        source_context: Optional[parsing.ParserContext],
+        ctx: context.ContextLevel) -> irast.Set:
+
+    assert isinstance(source, s_scalars.ScalarType)
+    enum_values = source.get_enum_values(ctx.env.schema)
+    assert enum_values is not None
+
+    if ptr_name not in enum_values:
+        raise errors.InvalidReferenceError(
+            f"'{source.get_displayname(ctx.env.schema)!r}' enum "
+            f"has no member called {ptr_name!r}; choose one from "
+            f"{set(enum_values)!r}",
```

Comment:
Not sure if listing all enum members is totally safe, because there can be lots of them.  Perhaps just do the levenstein thing?